### PR TITLE
✨ Consumer health check

### PR DESCRIPTION
**Pull request by RTLS**
https://www.loom.com/share/924329548e4045588765fb22df0db361

### CODE REVIEW

**Comment by RTLS** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/158?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#158** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/158?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#156** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#151** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/151?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/consumers/consumers.ex [hunk 1]
```diff
@@ -892,6 +894,34 @@ defmodule Sequin.Consumers do
...
+          id: :consumer_filters,
+          name: "Consumer filters",
+          status: :healthy,
+          error: nil
+        })
+
+        true
+
+      false ->
+        false
```
Comment (acco) [inline lib/sequin/consumers/consumers.ex:909]: filters still worked

File: lib/sequin/consumers/consumers.ex [hunk 2]
```diff
@@ -892,6 +894,34 @@ defmodule Sequin.Consumers do
...
+        name: "Consumer filters",
+        status: :unhealthy,
+        error:
+          Error.service(
+            code: :argument_error,
+            message: Exception.message(error)
+          )
+      })
+
+      false
```
Comment (acco) [inline lib/sequin/consumers/consumers.ex:924]: halt everything

File: lib/sequin/consumers_runtime/http_push_pipeline.ex [hunk 3]
```diff
@@ -55,9 +56,23 @@ defmodule Sequin.ConsumersRuntime.HttpPushPipeline do
 
     case push_message(http_endpoint, consumer_event.data, req_opts) do
       :ok ->
+        Health.update(consumer, %Health.Check{
```
Comment (acco) [inline lib/sequin/consumers_runtime/http_push_pipeline.ex:59]: debounce / cache

**Review by acco — APPROVED** [review]

**Pull request merged**